### PR TITLE
Add deprecation of namespace 'base' in favor of namespace 'numeric'

### PR DESCRIPTION
**Pull request by 2maz**
This commit allows to deprecate classes from numeric in the namespace base.
Deprecated classes can be removed by compiling with -DNUMERIC_DEPRECATE=1

Testing has been done with unit_tests and compiling slam/envire and running corresponding unit_tests there.

### CODE REVIEW

File: src/backwards/Histogram.hpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+It will be removed in future releases. \
+Adapt/Rename to numeric::Histogram instead. \
+Test your change by building base/numeric with -DNUMERIC_DEPRECATE=1"
+
+#include <algorithm>
+
+namespace base
+{
+
+template <class T>
```
Comment (doudou) [inline src/backwards/Histogram.hpp]: I think you can (and probably should) do

```
using numeric::Buckets
```

**Comment by 2maz** [comment]
updated #2

**Comment by doudou** [comment]
Got a comment, but somehow forgot to send it.

The way deprecation is announced here is annoying, as one will get the warnings even after he/she changed their code. The most practical solution would probably be to change the message to something that tells them how to make the warning go away (as opposed to only testing the changes). Moreover, the message should say explicitely that the define has to be set on the modified package (as opposed to base/numeric itself)

**Comment by 2maz** [comment]
Completely right -- changed the wording of the warning accordingly. Hope that suits better.

**Comment by doudou** [comment]
Looks good to me, but I would rather have a third pair of eyes looking at it from @rock-core/core

**Pull request merged**